agent/agent.go

```diff
@@ -487,11 +487,11 @@ func (a *Agent) runTask(ctx context.Context, tc *taskContext) (bool, error) {
 	defer a.killProcs(ctx, tc, false)
 	defer tskCancel()
 
-	heartbeat := make(chan string, 1)
-	go a.startHeartbeat(tskCtx, tskCancel, tc, heartbeat)
-
 	innerCtx, innerCancel := context.WithCancel(tskCtx)
 
+	heartbeat := make(chan string, 1)
+	go a.startHeartbeat(innerCtx, tskCancel, tc, heartbeat)
```

Comment:
Can we add a comment noting that this intentionally passes in the idle timeout context because we want to ensure that there is a deadline on how long the agent can spend handling idle timeout?

Reply:
i tried to simplify it? i hope it makes sense